src/structures/GuildChannel.js

```diff
@@ -35,37 +35,47 @@ class GuildChannel extends Channel {
      * @type {Guild}
      */
     this.guild = guild;
+
+    this.parentID = null;
   }
 
   _patch(data) {
     super._patch(data);
 
-    /**
-     * The name of the guild channel
-     * @type {string}
-     */
-    this.name = data.name;
+    if ('name' in data) {
+      /**
+       * The name of the guild channel
+       * @type {string}
+       */
+      this.name = data.name;
+    }
 
-    /**
-     * The raw position of the channel from discord
-     * @type {number}
-     */
-    this.rawPosition = data.position;
+    if ('position' in data) {
+      /**
+       * The raw position of the channel from discord
+       * @type {number}
+       */
+      this.rawPosition = data.position;
+    }
 
-    /**
-     * The ID of the category parent of this channel
-     * @type {?Snowflake}
-     */
-    this.parentID = data.parent_id || null;
+    if ('parent_id' in data) {
+      /**
+       * The ID of the category parent of this channel
+       * @type {?Snowflake}
+       */
+      this.parentID = data.parent_id;
+    }
 
-    /**
-     * A map of permission overwrites in this channel for roles and users
-     * @type {Collection<Snowflake, PermissionOverwrites>}
-     */
-    this.permissionOverwrites = new Collection();
-    if (data.permission_overwrites) {
-      for (const overwrite of data.permission_overwrites) {
-        this.permissionOverwrites.set(overwrite.id, new PermissionOverwrites(this, overwrite));
+    if ('permission_overwrites' in data) {
+      /**
+       * A map of permission overwrites in this channel for roles and users
+       * @type {Collection<Snowflake, PermissionOverwrites>}
+       */
+      this.permissionOverwrites = new Collection();
```

Comment:
This will make it be possibly undefined now, if at any point we create a channel with partial data